### DISCUSSION TITLE
9.0.0-beta.2 released

### DISCUSSION THREAD

**Discussion by Feiyang1** [firebase/firebase-js-sdk#4910] [General]
We just released 9.0.0-beta.2 with bug fixes and enhancements:

- Fixed a typing error in `@firebase/auth`. https://github.com/firebase/firebase-js-sdk/issues/4853
- Fixed an issue in the Nodejs build for `@firebase/database`
- Added missing method `isSupported()` for `@firebase/messaging-compat`
- Fixed a typing issue for `onSnapshot()`, which resulted in `unknown` snapshot data type in `@firebase/firestore`. https://github.com/firebase/firebase-js-sdk/issues/4812
- Made `@firebase/firestore` and `@firebase/database` work correctly with asynchronously loaded/initialized `@firebase/auth`
- Fixed broken CDN scripts
- Added an submodule `@firebase/messaging/sw` to expose service worker API
- Added es5 builds for `@firebase/firestore` and `@firebase/firestore/lite`

**Comment by YutaUra** [comment]
The methods such as `addDoc` in firebase v9, is this currently available?
I have used the following, but it does not work.
I get no error and the following code does not execute

```ts
import { initializeApp } from 'firebase/app'
import { getFirestore } from 'firebase/firestore'
import { addDoc, collection, serverTimestamp } from 'firebase/firestore'

const app = initializeApp({
  // ...
})
const db = getFirestore(app)
const res = await addDoc(collection(db, 'users'), { name: 'sample-user' }) // This is not working
console.log('After addDoc', res) // This code is not excuted

```

**Reply by YutaUra** [reply]
@Feiyang1 Can you help me with this?

**Reply by hsubox76** [reply]
Can you file an issue for this?

**Reply by YutaUra** [reply]
Thank you, @hsubox76. Then I will create a repository to reproduce it and post it as an Issue.
It's now midnight here, so I'll start working on it tomorrow :)

**Reply by YutaUra** [reply]
@hsubox76 I found the cause!
I'm using Recoil to develop, and Recoil freezes the object in development mode. In this case, I set the `dangerouslyAllowMutability` flag to prevent it from freezing, and this bug no longer occurs.

```ts
// src/store/firebase.ts
import { FirebaseApp, getApps, initializeApp } from 'firebase/app'
import { FirebaseFirestore, getFirestore } from 'firebase/firestore'
import { atom, selector } from 'recoil'

export const getAppDefault = () => {
  const apps = getApps()
  if (apps.length > 0) return apps[0]
  return initializeApp({
    // config
  })
}

export const FirebaseAppAtom = atom<FirebaseApp>({
  key: 'FirebaseAppAtom',
  default: getAppDefault(),
  dangerouslyAllowMutability: true, // Firebase App is not freezable object
})

export const FirebaseFirestoreSelector = selector<FirebaseFirestore>({
  key: 'FirebaseFirestoreSelector',
  get: ({ get }) => {
    const app = get(FirebaseAppAtom)
    const db = getFirestore(app)
    return db
  },
  // If `dangerouslyAllowMutability` is false, it doesnt work
  dangerouslyAllowMutability: true, // Firebase Firestore is not freezable object
})

```

The only thing that was a little weird was that Recoil would give me an error message if it could not freeze, but it did not. Maybe it has something to do with firebase?

If you are interested in this, please see this. https://firestore-v9-not-working.web.app/

This site is fully working. (It says NOT WORKING in the title, but that's a lie because I found a way to fix it while I was trying to reproduce it. Sorry about that.
If you follow the instructions, clone it, and set `dangerouslyAllowMutability` to false, as in the comment above, you will see that it instantly stops working.

And should I make an Issue about this?

**Reply by Feiyang1** [reply]
I'm not familiar with recoil. Does it freeze all objects in your app including 3p libraries? 
Yeah, can you please open an issue? let's continue the conversation there.
